mobile/src/store/uam/reducer.ts

```diff
@@ -11,75 +10,70 @@ import { signOut } from '../auth/actions';
+  groups: GroupsItemResponseDto[];
+  users: UsersGetResponseDto[];
+  usersTotalCount: number;
+  groupsTotalCount: number;
   userDeleteDataStatus: DataStatus;
   groupDeleteDataStatus: DataStatus;
 };
 
 const initialState: State = {
-  dataStatus: DataStatus.IDLE,
-  groups: {
-    items: [],
-    total: 0,
-  },
-  users: {
-    items: [],
-    total: 0,
-  },
+  groupsDataStatus: DataStatus.IDLE,
+  usersDataStatus: DataStatus.IDLE,
+  users: [],
+  usersTotalCount: 0,
+  groups: [],
+  groupsTotalCount: 0,
   userDeleteDataStatus: DataStatus.IDLE,
   groupDeleteDataStatus: DataStatus.IDLE,
 };
 
 const reducer = createReducer(initialState, (builder) => {
   builder.addCase(getGroups.pending, (state) => {
-    state.dataStatus = DataStatus.PENDING;
+    state.groupsDataStatus = DataStatus.PENDING;
   });
   builder.addCase(getGroups.fulfilled, (state, action) => {
-    state.dataStatus = DataStatus.FULFILLED;
-    state.groups = action.payload;
+    state.groupsDataStatus = DataStatus.FULFILLED;
+    state.groups = action.payload.items;
+    state.groupsTotalCount = action.payload.total;
   });
   builder.addCase(getGroups.rejected, (state) => {
-    state.dataStatus = DataStatus.REJECTED;
+    state.groupsDataStatus = DataStatus.REJECTED;
   });
 
   builder.addCase(deleteGroup.pending, (state) => {
     state.groupDeleteDataStatus = DataStatus.PENDING;
   });
   builder.addCase(deleteGroup.fulfilled, (state, { payload }) => {
-    const items = state.groups.items.filter((item) => item.id !== payload);
-    state.groups = {
-      items: items,
-      total: items.length,
-    };
+    state.groups = state.groups.filter((item) => item.id !== payload);
+    state.groupsTotalCount = --state.groupsTotalCount;
     state.groupDeleteDataStatus = DataStatus.FULFILLED;
   });
   builder.addCase(deleteGroup.rejected, (state) => {
     state.groupDeleteDataStatus = DataStatus.REJECTED;
   });
 
   builder.addCase(getUsers.pending, (state) => {
-    state.dataStatus = DataStatus.PENDING;
+    state.usersDataStatus = DataStatus.PENDING;
   });
   builder.addCase(getUsers.fulfilled, (state, action) => {
-    state.dataStatus = DataStatus.FULFILLED;
-    state.users = action.payload;
+    state.usersDataStatus = DataStatus.FULFILLED;
+    state.users = action.payload.items;
+    state.usersTotalCount = action.payload.total;
   });
   builder.addCase(getUsers.rejected, (state) => {
-    state.dataStatus = DataStatus.REJECTED;
+    state.usersDataStatus = DataStatus.REJECTED;
   });
 
   builder.addCase(deleteUser.pending, (state) => {
     state.userDeleteDataStatus = DataStatus.PENDING;
   });
   builder.addCase(deleteUser.fulfilled, (state, { payload }) => {
-    const items = state.users.items.filter((item) => item.id !== payload);
-
-    state.users = {
-      items: items,
-      total: items.length,
-    };
+    state.users = state.users.filter((item) => item.id !== payload);
+    state.usersTotalCount = --state.usersTotalCount;
```

Comment:
```suggestion
    state.usersTotalCount = state.users.length;
```

Reply:
https://user-images.githubusercontent.com/89215185/185969646-a33831df-be59-4de6-a1e0-f95be5db37b4.mp4

@Liza-Veis, look at the app behavior on using users length. If it's ok, i'll left it!

Reply:
ok, let's leave it as it is